File: IOS/LuaViewSDK/Lua/Local/Net/NetworkManager.m
//
//  NetworkManager.m
//  LuaViewSDK
//
//  Created by 王金东 on 17/2/6.
//  Copyright © 2017年 王金东. All rights reserved.
//

#import "NetworkManager.h"


static id<NetworkDelegate> netWorkDelegate;

@implementation NetworkManager

+ (void)register:(id<NetworkDelegate>) delegate {
    netWorkDelegate = delegate;
}

+ (void)post:(NSString *)url
      params:(NSDictionary *)params
    callBack:(LuaCallBack)callBack {
    [netWorkDelegate post:url params:params callBack:callBack];
}

+ (void)downImage:(NSString *)url callBack:(LuaImageCallBack)callBack {
    [netWorkDelegate downImage:url callBack:callBack];
}

@end

File: Android/LuaViewSDK/app/src/main/java/com/core/lua/local/net/NetworkManager.java
package com.core.lua.local.net;

import java.util.Map;

/**
 * Created by JD on 2017/2/6.
 */
public class NetworkManager {

    public static NetworkInterface netWorkInterface;

    public static void register(NetworkInterface netWork){
            netWorkInterface = netWork;
    }

    public static void post(String url, Map params, NetworkInterface.NetworkCallBack callBack) {
        netWorkInterface.post(url, params, callBack);
    }

    public static void downImage(String url,NetworkInterface.ImageCallBack callBack) {
        netWorkInterface.downImage(url,callBack);
    }
}
